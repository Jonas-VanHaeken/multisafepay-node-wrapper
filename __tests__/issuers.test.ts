import axios from 'axios';
import Issuers from '../src/resources/issuers';
import issuersResponse from '../__stubs__/issuers.stub.json';
import MockAdapter from 'axios-mock-adapter';

const mock = new MockAdapter(axios);

describe('issuers', () => {
  let issuers: any;
  beforeEach(() => {
    issuers = new Issuers(axios.create());
  });
  it('check endpoint', () => {
    // TODO import enum endpoint
    expect(Issuers.endpoint).toBe('issuers/ideal');
  });

  it('get ideal with success', async () => {
    mock.onGet('/issuers/ideal').reply(200, issuersResponse);

    const resp = await issuers.get();

    expect(resp).toMatchSnapshot();
  });

  it('get error sending wrong endpoint', async () => {
    const error = {
      error_code: 1036,
      error_info: 'Invalid iDEAL issuer ID',
    };
    const newEndpoint = 'issuers/hello';
    Issuers.endpoint = newEndpoint;
    mock.onGet(newEndpoint).reply(404, error);
    const resp = await issuers.get();

    expect(resp).toEqual(error);
  });
});
