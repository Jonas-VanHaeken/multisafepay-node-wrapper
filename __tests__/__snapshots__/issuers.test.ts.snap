// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`issuers get ideal with success 1`] = `
Object {
  "message": Object {
    "data": Array [
      Object {
        "code": "3151",
        "description": "Test bank",
      },
      Object {
        "code": "0031",
        "description": "ABN AMRO",
      },
      Object {
        "code": "0751",
        "description": "SNS Bank",
      },
      Object {
        "code": "0721",
        "description": "ING",
      },
      Object {
        "code": "0021",
        "description": "Rabobank",
      },
      Object {
        "code": "0761",
        "description": "ASN Bank",
      },
      Object {
        "code": "0771",
        "description": "Regio Bank",
      },
      Object {
        "code": "0511",
        "description": "Triodos Bank",
      },
      Object {
        "code": "0161",
        "description": "Van Lanschot Bankiers",
      },
      Object {
        "code": "0801",
        "description": "Knab",
      },
      Object {
        "code": "4371",
        "description": "bunq",
      },
      Object {
        "code": "1234",
        "description": "Moneyou",
      },
      Object {
        "code": "1235",
        "description": "Handelsbanken",
      },
    ],
    "success": true,
  },
}
`;
